README.md

```diff
@@ -57,25 +63,35 @@ jobs:
   clang-tidy:
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v2
-      - name: Build project and/or unit tests
-        run: ./your-build-script.sh
-      - name: Run clang-tidy
-        run: ./your-clang-tidy-script.sh --fixes-path fixes.yaml
-      - name: Run clang-tidy-pr-comments action
-        uses: platisd/clang-tidy-pr-comments@master
-        with:
-          # The GitHub token (or a personal access token)
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          # The path to the clang-tidy fixes generated previously
-          clang_tidy_fixes: fixes.yaml
-          # Optionally set to true if you want the Action to request
-          # changes in case warnings are found
-          request_changes: true
-          # Optionally set the number of comments per review
-          # to avoid GitHub API timeouts for heavily loaded
-          # pull requests
-          suggestions_per_comment: 10
+    - uses: actions/checkout@v2
+    - name: Install clang-tidy
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y clang-tidy
+    - name: Prepare compile_commands.json
+      run: |
+        cmake -B build -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+    - name: Create results directory
+      run: |
+        mkdir clang-tidy-result
+    - name: Analyze
+      run: |
+        git diff -U0 HEAD^ | clang-tidy-diff -p1 -path build -export-fixes clang-tidy-result/fixes.yml
+    - name: Run clang-tidy-pr-comments action
+      if: ${{ hashFiles( 'clang-tidy-result/fixes.yml' ) != '' }}
```

Comment:
Is this making sure to run the Action only if there are fixes?
Doesn't the Action already handle that internally? Since this is an example, isn't it better to keep it simpler rather than optimized?

Reply:
It does - if this file exists, but is empty. But if it's not generated at all (and this can happen if the source files were changed in such a way that the code was only deleted, but not added or changed, like [here](https://github.com/ihhub/fheroes2/pull/3895)), Action just fails. So we need to run this Action only if the `fixes.yml` is present.

Reply:
Interesting edge-case! I am a little split over whether it should be excluded from the example to favor simplicity or modify the Action and let it handle missing files (i.e. not fail).

On the other hand, I am OK-ish including it in the example, however some more explanation is necessary (as to _why_ we need this check) either as a comment or in the `name` of the step.

What do you think?

Reply:
I can add a comment clarifying why this check is needed. I didn't modify the Action itself because there may be typos in the yml file name or path for example, and in this case it's better to catch it early (failing Action is preferred in this case).

Reply:
Sounds good!